test/benchmark/sframe_bench_aggregate.cpp

```diff
@@ -0,0 +1,287 @@
+/* Copyright © 2017 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+#include <atomic>
+#include <core/storage/sframe_data/groupby_aggregate.hpp>
+#include <core/storage/sframe_data/groupby_aggregate_operators.hpp>
+#include <core/storage/sframe_data/sframe.hpp>
+#include <core/storage/sframe_data/sframe_constants.hpp>
+#include <iostream>
+#include <thread>
+#include <random>
+
+namespace turi {
+
+using row_gen_func_t = std::function<std::vector<flexible_type>(size_t)>;
+
+/* generate sframe */
+/* I need to make this a fixture */
+static sframe bench_test_sframe_generator(
```

Comment:
Use random sframe generator code in <core/storage/sframe_data/testing_utils.hpp> -- that would be easier to use. We don't want to duplicate code if we have good reason not to.